### ISSUE TITLE
Summary page for all repos status'

### ISSUE THREAD

**Issue by bradleyfalzon** [bradleyfalzon/gopherci#17]
As a user, I should be able to see an overview of all my repos and see what issues I might need to solve. This could be:

- Repo Name
- Protected Branches (if I want all branches protected, show me those that aren't)
- License (alert if there's a missing or incorrect license)
- Outstanding issues
- Outstanding PRs
- Results from tools (could just be a sum of issues from all tools or breakdown per tool)

This would need the ability to scan an entire repo without revgrep filters, and then continue to update it when there's new commits. Providing a single page overview of the health of all your repos based on the rules you've set.

**Comment by bradleyfalzon** [comment]
I'd like to reduce this scope to just: `Results from tools (could just be a sum of issues from all tools or breakdown per tool)`, perhaps when #69 is implemented that could make this possible (if we store the number or list of issues found per repo).

**Comment by shurcooL** [comment]
I reacted with 👍 but want to add more details.

I think this could be absolutely great and helpful.

Right now, I know of GopherCI mostly from the status messages it leaves on PRs, etc. Recently, I know it started leaving comments on commits, etc., although I haven't seen that in action yet because I haven't pushed commits recently.

But there's a missing part of GopherCI that I notice, compared to other CI systems I've used, and that is an "index page" of sorts. A page where you can see an overview of all the repos and their checks, statuses, etc.

I feel a page like that is an import part of the "identity" of the CI system and helps communicate what it does. It provides a way to glance over everything happening. It's also a way to see individual repositories and click on them to see more details.

Without an index page like that, it's hard to do those things. Which is why I think, if done well, it can be very benefitial.

For reference, here's a screenshot of the Travis CI index page for me. If that page were removed, my Travis CI experience would be significantly negatively impacted.

![image](https://user-images.githubusercontent.com/1924134/29286837-734cf2f0-8101-11e7-9787-cd7658fa01e1.png)

**Comment by shurcooL** [comment]
Quick note, there are some factors to consider when making the comparison above. I use Travis CI to ensure basic functionality and need everything to be passing [95%+ of the time in a year](https://github.com/shurcooL/SLA#uptime). If one of my repositories has a Travis build failure, that's a very serious issue that I need to address promptly.

On the other hand, I use CircleCI right now as a linter of sorts. It may spot issues that I missed locally and let me know about them. But the reports are informative only, there is no "success"/"failure" status.

So, such an index page with ability to glance over many repos may not be as missing critical to GopherCI as it is for Travis CI (which is why you are able to get away without having one so far), but I still think it can be very helpful.

**Comment by bradleyfalzon** [comment]
I agree it will be a useful feature @shurcooL and thanks for the input, this helps prioritise the work. I'm getting a better idea as to how this can work, especially when combined with #69 now.

Further notes, we should probably do this on a per branch basis.

Also, keep in mind, there's a goal that you should be able to enable testing of repositories too, so viewing all repositories with failing tests may be helpful (when testing is as simple as `go test ./...`). So we may continue with a success, warning, failure status model.
